changelogutils/changelogdocutils/merged_release.go

```diff
@@ -0,0 +1,201 @@
+package changelogdocutils
+
+import (
+    "context"
+    "encoding/json"
+    "fmt"
+    "github.com/google/go-github/v32/github"
+    "github.com/solo-io/go-utils/githubutils"
+    . "github.com/solo-io/go-utils/versionutils"
+    "regexp"
+)
+
+type DependencyFn func(*Version) (*Version, error)
+
+type Options struct {
+    // Number of versions from the most previous version to display in changelog
+    NumVersions int
+    // Maximum version changelog to display on this changelog
+    MaxVersion *Version
+    // Minimum version changelog to display on this changelog
+    MinVersion *Version
+    ProjectName,
+    RepoOwner,
+    MainRepo,
+    DependentRepo string
```

Comment:
The reason I chose `Main` over `Enterprise` is because this same struct is used for the `minor_release` changelog, which doesn't use the open source repo since it doesn't do any multi-repo merging.